### ISSUE TITLE
Able to modify built-in flow

### ISSUE THREAD

**Issue by upfluentpatrickpilch** [keycloak/keycloak#15536]
### Area

admin/api

### Describe the bug

I am able to add a sub-flow to a built-in flow (e.g. registration). See bottom of screenshot:

![image](https://user-images.githubusercontent.com/97203210/202324055-5ee7e703-5a57-4f92-8ca7-ce7509601973.png)

### Version

20.0.0

### Expected behavior

Since I am unable to add steps to a "Built-in" flow (resulting in "Could not update flow: It is illegal to add execution to a built in flow"), I assume the built-in flows are meant to be immutable. 

Therefore, I would not expect to be able to add a sub-flow.

### Actual behavior

I am able to add a sub-flow to a built-in flow.

### How to Reproduce?

1. Go to Authentication in admin UI.
2. Select "registration (Built-in)" flow.
3. Click "Add sub-flow"
4. Leave the defaults with the exception of providing a required name.
5. Click "Add"

**Comment by mposolda** [comment]
Also seen this recently.

I am adding to Keycloak 22 for now, but it is possible that we may need to postpone (this is not strictly a blocker for Keycloak 22).

It may turn to be only the issue in UI (new admin console ) - at least it is strange to show buttons "Add step" and "Add sub-flow" for the built-in flows, which we want to be immutable. But it's possible it is issue for admin REST API as well.

**Comment by rmartinc** [comment]
@upfluentpatrickpilch You are right, that particular endpoint (adding a sub-flow to a built in flow) is allowed while the other endpoints returns an error. It's a minor issue but I will send a PR to fix it.

@mposolda Do you think that makes sense to also add a UI issue to the admin console to hide the buttons when the flow is built in?
